src/VisualStudio/IntegrationTest/New.IntegrationTests/VisualBasic/BasicGoToImplementation.cs

```diff
@@ -37,9 +41,35 @@ Implements IGoo
 End Interface", HangMitigatingCancellationToken);
             await TestServices.Editor.PlaceCaretAsync("Interface IGoo", charsOffset: 0, HangMitigatingCancellationToken);
             await TestServices.Editor.GoToImplementationAsync(HangMitigatingCancellationToken);
+
+            string identifierWithCaret;
+            var activeCaption = await TestServices.Shell.GetActiveWindowCaptionAsync(HangMitigatingCancellationToken);
             var dirtyModifier = await TestServices.Editor.GetDirtyIndicatorAsync(HangMitigatingCancellationToken);
+            if (activeCaption == $"FileImplementation.vb{dirtyModifier}")
+            {
+                // The navigation completed synchronously; no further action necessary
+                identifierWithCaret = "Implementation$$";
+            }
+            else
+            {
+                // The navigation completed asynchronously, so navigate to the first item in the results list
+                Assert.Equal($"'IGoo' implementations - Entire solution", activeCaption);
+                var results = await TestServices.FindReferencesWindow.GetContentsAsync(HangMitigatingCancellationToken);
+                AssertEx.EqualOrDiff(
+                    $"<unknown>: Class Implementation",
+                    string.Join(Environment.NewLine, results.Select(result => $"{result.GetItemOrigin()?.ToString() ?? "<unknown>"}: {result.GetText()}")));
+                results[0].NavigateTo(isPreview: false, shouldActivate: true);
+
+                // It's not clear why this delay is necessary. Navigation operations are expected to fully complete as part
+                // of one of the above waiters, but GetActiveWindowCaptionAsync appears to return "Program.cs" (the previous
+                // window caption) for a short delay after the above complete.
```

Comment:
def should talk to eidtor.  from your descriptions, it sounds like tehy're async updating tab titles.  but that's def problematic for tests :(

Reply:
Can this at least link to a tracking bug or that other PR or something?